src/SW_Weather.c

```diff
@@ -1177,13 +1656,103 @@ void _read_weather_hist(
-        yearWeather->temp_max[doy] = tmpmax;
-        yearWeather->temp_min[doy] = tmpmin;
-        yearWeather->ppt[doy] = ppt;
+        yearWeather->temp_max[doy] = weathInput[dailyInputIndices[TEMP_MAX]];
+        yearWeather->temp_min[doy] = weathInput[dailyInputIndices[TEMP_MIN]];
+        yearWeather->ppt[doy] = weathInput[dailyInputIndices[PPT]];
 
         // Calculate average air temperature if min/max not missing
-        if (!missing(tmpmax) && !missing(tmpmin)) {
-          yearWeather->temp_avg[doy] = (tmpmax + tmpmin) / 2.0;
+        if (!missing(weathInput[dailyInputIndices[TEMP_MAX]]) &&
+            !missing(weathInput[dailyInputIndices[TEMP_MIN]])) {
+
+          yearWeather->temp_avg[doy] = (weathInput[dailyInputIndices[TEMP_MAX]] +
+                                        weathInput[dailyInputIndices[TEMP_MIN]]) / 2.0;
+        }
+
+        if(!use_cloudCoverMonthly) {
+            if(dailyInputFlags[CLOUD_COV]) {
+                yearWeather->cloudcov_daily[doy] = weathInput[dailyInputIndices[CLOUD_COV]];
+            }
+        }
+
+        if(!use_windSpeedMonthly) {
+            if(dailyInputFlags[WIND_SPEED]) {
+                yearWeather->windspeed_daily[doy] = weathInput[dailyInputIndices[WIND_SPEED]];
+            } else if(dailyInputFlags[WIND_EAST] && dailyInputFlags[WIND_NORTH]) {
+                yearWeather->windspeed_daily[doy] = sqrt(squared(weathInput[dailyInputIndices[WIND_EAST]]) +
+                                                         squared(weathInput[dailyInputIndices[WIND_NORTH]]));
+            }
+        }
+
+        // Check to see if monthly humidity values are being used
+        // If not, skip entire code block to elimiate unecessary checking of many flags
+        if(!use_humidityMonthly) {
+            if(hasMaxMinRelHumid) {
+                yearWeather->r_humidity_daily[doy] = (weathInput[dailyInputIndices[REL_HUMID_MAX]] +
+                                                      weathInput[dailyInputIndices[REL_HUMID_MIN]]) / 2;
+
+            } else if(dailyInputFlags[REL_HUMID]) {
+                yearWeather->r_humidity_daily[doy] = weathInput[dailyInputIndices[REL_HUMID]];
+
+            } else if(dailyInputFlags[SPEC_HUMID]) {
+                // Specific Humidity (Bolton 1980)
+                es = (6.112 * exp(17.67 * yearWeather->temp_avg[doy]) /
+                     (yearWeather->temp_avg[doy] + 243.5));
+
+                e = (weathInput[dailyInputIndices[SPEC_HUMID]] * 1013.25) /
+                    (.378 * weathInput[dailyInputIndices[SPEC_HUMID]] + .622);
+
+                relHum = e / es;
+                relHum = max(0., relHum);
+
+                yearWeather->r_humidity_daily[doy] = min(100., relHum);
+
+                // Check if a calculation of actual vapor pressure occurred with
+                // bad temperature values (SW_MISSING)
+                if(missing(yearWeather->temp_avg[doy])) {
+
+                    yearWeather->r_humidity_daily[doy] = SW_MISSING;

```

Comment:
Why calculate relative humidity from specific humidity if we are going to override it anyhow? Why not an if-else?
```
if(!missing(yearWeather->temp_avg[doy])) {
  // do the calculation
} else {
  // set to missing
}
```